src/static/riot/profile/profile.tag

```diff
@@ -0,0 +1,362 @@
+        }
+
+        self.add_email = function (data) {
+            data.preventDefault()
+            let email_address = $(self.refs.email_address).val()
+            if (email_address) {
+                CHAHUB.api.add_email(self.opts.user_pk, email_address)
+                    .done(function (data) {
+                        toastr.success('Email address added. A verification email has been sent')
+                        self.update_user()
+                    })
+                    .fail(function (response) {
+                        toastr.error('Could not add email address')
+                    })
+
+            }
+
+            $(self.refs.email_modal).modal('hide')
+        }
+
+        self.delete_email = function (email_pk) {
+            if (confirm('Are you sure you want to delete this email address?')) {
+                CHAHUB.api.delete_email(self.opts.user_pk, email_pk)
+                    .done(function () {
+                        toastr.success('Email address deleted')
+                        self.update_user()
+                    })
+                    .fail(function () {
+                        toastr.error('Could not delete email address')
+                    })
+            }
+        }
+
+        self.make_primary_email = function (email_pk) {
+            if (confirm("Are you sure you want to make this your primary email account?")) {
+                CHAHUB.api.make_primary_email(self.opts.user_pk, email_pk)
+                    .done(function () {
+                        toastr.success('Primary Account Changed')
+                        self.update_user()
+                    })
+                    .fail(function () {
+                        toastr.error('Error changing primary email')
+                    })
+            }
+        }
+
+        self.resend_verification_email = function (email_pk) {
+            CHAHUB.api.resend_verification_email(self.opts.user_pk, email_pk)
+                .done(function () {
+                    toastr.success('Email sent!')
+                })
+                .fail(function () {
+                    toastr.error('Could not resend email')
+                })
+        }
+
+        self.delete_profile = function () {
+            $(self.refs.delete_profile_modal).modal('hide')
+            CHAHUB.api.delete_profile(self.opts.user_pk, self.selected_profile_id)
+                .done(function () {
+                    toastr.success('Profile deleted')
+                    self.update_user()
+                })
+                .fail(function () {
+                    toastr.error('Error deleting profile')
+                })
+            self.selected_profile_id = undefined
+
+        }
+
+        self.delete_user = function () {
+            CHAHUB.api.delete_user(self.opts.user_pk)
+                .done(function (data) {
+                    toastr.success('Successfully Deleted User')
+                    location = URLS.HOME
+                })
```

Comment:
what if this one fails, some error message should appear?

Reply:
ahh, yup, good catch. Adding a `.fail()`